### PR TITLE
More letters testing

### PR DESCRIPTION
**Pull request by rachelwong**
**_Update:  All tests now passing. Please note Commands.js & moreLetters.spec.js._** 

----
Apologies - removed last PR due to minor edits detected. I should have mentioned this to you when we spoke, I am very sorry about this.

Issues:
- tests not passing 
- unable to successfully pull sessionStorage data, display on DOM, assert correct result after button click. In other words, not able to reproduce user scenario using cypress commands to test the scoring, generating letters capabilities of my script.js. I don't know what I am not doing correctly. 

Attempted:
- inserting sessionStorage hand (stubbing)
- used a variety of assertion methods including: 
* expect().to.have.lengthOf(), 
* cy.get(hand).should("have.value", hand), 
* cy.get(hand).contains(hand), 
* cy.get(hand).should("contain", hand). 

This usually result in the "hand" (being displayed on the DOM) to have a value of empty string ("") even though there is something being displayed. Thus failing tests.

For context, in my **script.js**, the hand array is being displayed on the DOM via .innerHTML.

If you have any feedback at all in relation to how I should manage PRs, developing on feature branches.

### CODE REVIEW

File: cypress/integration/moreLetters.spec.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+describe("More letters button", () => {
+  beforeEach(() => {
+    // deep clean
+    cy.window().then((win) => {
+      win.sessionStorage.clear()
+    })
+    cy.clearLocalStorage()
+    // start new game
+    cy.visit("http://127.0.0.1:5501")
```
Comment (dan-dekel) [inline cypress/integration/moreLetters.spec.js]: Let's point to netlify env

File: cypress/integration/moreLetters.spec.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    // arrange initial state - seed new hand
+    const fullHand = "inertia".split("")
+    sessionStorage.setItem("hand", JSON.stringify(fullHand))
+
+    // action - submit empty space (no word)
+    // action - no change to hand in display
+    // action - no change to sessionStorage (hand fully visible with 7 characters)
+    cy.get('[cy-data="newHandBtn"]').should("exist").click()
+    const emptyValue = " "
+    cy.get('[cy-data="wordInputForm"]').type(emptyValue)
```
Comment (dan-dekel) [inline cypress/integration/moreLetters.spec.js]: Hey Rachel, something I just noticed - Cypress best practice does suggest we use insulted attributes on UI elements. I think you just switched the prefix around. i.e. '[cy-data=...]' should be '[data-cy-=...]'.
Here's the reference in the docs (I'm sure you've already seen this):
https://docs.cypress.io/guides/references/best-practices.html#How-It-Works

File: cypress/integration/moreLetters.spec.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      const placeholderInput = "tin"
+      cy.get('[cy-data="wordInputForm"]').type(placeholderInput)
+      cy.get('[cy-data="submitBtn"]').click()
+
+      // action - get more letters clicked
+      cy.get('[cy-data="newLettersBtn"]').should("be.visible").click()
+
+      // assert - 7 letters in sessionStorage hand
+      const checkHand = JSON.parse(sessionStorage.getItem("hand"))
+      console.log('checkHand:', checkHand)
```
Comment (dan-dekel) [inline cypress/integration/moreLetters.spec.js]: we want to remove console.log from our final code

File: cypress/integration/moreLetters.spec.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  context("current hand has 0 letters", () => {
+    beforeEach(() => {
+
+    })
+  })
+
+  it("hand has 0 letters, can get 7 letters", () => {
+    // arrange initial state - seed no letters in hand (inertia)
+    const emptyHand = []
+    sessionStorage.setItem("hand", JSON.stringify(emptyHand))
```
Comment (dan-dekel) [inline cypress/integration/moreLetters.spec.js]: We've got a few occurrences of this method used repetitively. It's a good opportunity to refactor and move it to its own function (maybe as a command?)

File: cypress/integration/moreLetters.spec.js (hunk 1 above)
Comment (rachelwong) [reply cypress/integration/moreLetters.spec.js]: Done

**Comment by rachelwong** [comment]
Thank you Dan for the comments, changes made as instructed. 

Re-clarified the test cases for moreLetters - all tests passing. Files to note are:
- Commands.js
- moreLetters.spec.js

File: cypress/integration/moreLetters.spec.js (hunk 2 above)
Comment (rachelwong) [reply cypress/integration/moreLetters.spec.js]: done

File: cypress/integration/moreLetters.spec.js (hunk 3 above)
Comment (rachelwong) [reply cypress/integration/moreLetters.spec.js]: done

File: cypress/integration/moreLetters.spec.js (hunk 4 above)
Comment (rachelwong) [reply cypress/integration/moreLetters.spec.js]: done. **cy.seedHand()**

**Pull request merged**